DevelopmentTools/Rakefile

```diff
@@ -143,24 +143,65 @@ task release: %i[ensure_git_clean chdir] do
 
   tag_name = 'v' + nextversion
 
-  github = MyGithub.github
-  release = github.repos.releases.create owner, repository,
-    tag_name: tag_name,
-    target_commitish: 'master',
-    name: "Release for version #{tag_name}",
-    draft: true,
-    prelease: false
-
   artifact_targets.each do |tar|
     puts "--- #{tar} ---"
     Rake::Task['build'].invoke(tar)
     Rake::Task['build'].reenable
   end
 
+  release_notes = "# Release notes:\n"
+
+  File.open('CHANGELOG.md', 'r') do |changelog|
+    release_notes += changelog
+                     .read
+                     .gsub(/\\\* \*This Change Log was automatically generated by.*\*/, '')
+                     .split(/## \[.*\]\(.*\) \(.*\).*\n\s*\[Full Changelog\]\(.*\).*\n/)[1].chomp
+  end
+
+  github = MyGithub.github
+  release = github.repos.releases.create owner, repository,
+                                         tag_name: tag_name,
+                                         target_commitish: 'master',
+                                         name: "Version #{tag_name}",
+                                         body: release_notes,
+                                         draft: true,
+                                         prelease: false
+
+  Rake::Task['artifacts'].invoke(release['id'])
+end
+
+desc 'Add artifacts to the release'
+task :artifacts, [:id] => [:chdir] do |_t, args|
+  id = args['id']
+  UI.user_error!('Release id is required to attach artifacts to it') unless id
+
+  require 'net/http'
   Dir['target/*.unitypackage'].each do |artifact|
-    github.repos.releases.assets.upload owner, repository, release['id'], artifact,
-      name: File.basename(artifact),
-      content_type: 'application/octet-stream'
+    puts "Uploading #{artifact}..."
+    uri = URI.parse("https://uploads.github.com/repos/#{owner}/#{repository}/releases/#{id}/assets?name=#{File.basename(artifact)}")
+    Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
+      request = Net::HTTP::Post.new(
+        uri.request_uri,
+        initheader = {
+          'Content-Type' => 'application/zip',
+          'Authorization' => 'token ' + ENV['UPLIFT_GITHUB_TOKEN']
+        }
+      )
+      File.open(artifact, 'rb') do |f|
+        request.body = f.read
+      end
+      response = http.request request
+      unless response.code.between?('200', '299')
+        puts "Could not upload #{artifact} to #{uri}"
+        puts "Response failed with code #{response.code}"
+        puts response.body
+      end
+    end
+    # FIXME
```

Comment:
FIXME?

Reply:
Imo it would be better if we could use the GihubAPI gem and not have to create our own requests. Especially because it could be useful for the future.